### PR TITLE
use custom setter methods (cherry-pick beta -> master), bump connect-js version

### PR DESCRIPTION
**Pull request by kaiying-stripe**

### CODE REVIEW

File: src/useCreateComponent.tsx [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+import {
+  ConnectElementTagName,
+  ConnectHTMLElementRecord,
+} from '@stripe/connect-js';
 
-export const useCreateComponent = (
-  tagName: ConnectElementTagName
-): {wrapper: JSX.Element; component: HTMLElement | null} => {
-  const [component, setComponent] = React.useState<HTMLElement | null>(null);
+export const useCreateComponent = <T extends ConnectElementTagName>(
```
Comment (jorgea-stripe) [inline src/useCreateComponent.tsx:9]: @kaiying-stripe  Did you mean to update the connect.js version in the package.json?

**Review by jorgea-stripe — CHANGES_REQUESTED** [review]

File: src/Components.tsx [hunk 2]
```diff
@@ -18,26 +18,21 @@ export const ConnectAccountOnboarding = ({
   const {wrapper, component: onboarding} =
     useCreateComponent('account-onboarding');
 
-  useAttachAttribute(
+  React.useEffect(() => {
+    if (!onboarding) return;
+    onboarding.setRecipientTermsOfServiceUrl(recipientTermsOfServiceUrl);
```
Comment (jorgea-stripe) [inline src/Components.tsx:23]: It is a bit odd to call all of the setters. Does it make sense to instead have separate useEffects like we had before? We can have a hook wrappers that do that as well, like:

```
useUpdateWithSetter(onboarding, fullTermsOfServiceUrl, (elem, val) => elem.setFullTermsOfServiceUrl(val));
```

**Review by jorgea-stripe — CHANGES_REQUESTED** [review]

File: src/useCreateComponent.tsx (hunk 1 above)
Comment (kaiying-stripe) [reply src/useCreateComponent.tsx:9]: Yeah, I forgot to push that 😅

File: package.json [hunk 3]
```diff
@@ -46,7 +46,7 @@
     "@babel/preset-flow": "7.18.6",
     "@babel/preset-react": "7.18.6",
     "@rollup/plugin-replace": "^2.3.1",
-    "@stripe/connect-js": "2.0.3",
+    "@stripe/connect-js": "3.0.1",
```
Comment (jorgea-stripe) [inline package.json:49]: @kaiying-stripe make sure to also bump the `peerDependencies`!

**Review by jorgea-stripe — APPROVED** [review]
@kaiying-stripe PR looks good, but we are missing the `peerDependencies` bump

**Pull request merged**